etcdmgmt/etcd-mgmt.go

```diff
@@ -35,6 +36,11 @@ func StartEtcd() error {
 		"-initial-advertise-peer-urls", initialAdvPeerUrls,
 		"--initial-cluster", "default="+listenPeerUrls)
 
+	// Don't kill chlid process (etcd) upon ^C (SIGINT) of main glusterd process
```

Comment:
>> Point 1 : Currently glusterd2 daemon itself is not daemonized and hence once you execute it it doesnt return back to the prompt and with that you bring in interrupts, however this wouldn't be true as we have to daemonize it (this is how existing glusterd also behave). So if your observation is that etcd doesnt go down in SIGKILL or SIGTERM we dont need this change at all as we can run the current glusterd2 instance in background to avoid the command prompt to get stuck.

>>>Yes, that's true. When glusterd2 daemon itself will daemonized then we don't need to execute ^C and without this change it will work. but its a safe to don't include etcd daemon id in parent process group id.

I am just wondering why SIGINT is behaving differently here. Do you have any background?

 >> Point 2 : Being a programmer I expect these basic logics to be thought out before sending the changes. A github issue will highlight what is the problem we are tryung to solve but it will never say how are you going to achieve it. Starting etcd instance without checking whether an already etcd process is running or not is a mistake, no matter whatever reason you provide and I dont want to get into that further.

>>>That i will take care in my next PR

Good that we have an agreement here

>> Point 3: Word 'daemonize' has some significant meaning. Daemonizing a process doesn't only mean changing the a single attribute of a process. Again to repeat daemonize itself explains what needs to be done.

>>> I didn't expect that we are expecting daemonizing framework by end of this month. If i remember correctly by this EOM we just need to avoid ectd killing upon ^C. 

If that's the case use a different issue number or atleast don't use the term 'daemonize'

>>    You were partially right that if a SIGKILL/SIGTERM is sent to the process, child process remains as it is but thats only true for a foreground process. If glusterd2 is run in background its not correct.

>>>Again i will ask you that did your etcd daemon killed when you run glusterd2 process in background and if yes i would like to know the process of killing because in my system it didn't killed.

This is a mistake from end. It doesn't happen like that. You are correct.

So to summarize following changes/actions are required:

Add a check of whether etcd is running or not before starting etcd process to avoid any defunct instances
Address @kshlm's comments
Remove the word 'daemonize' from the commit message and change the heading of the issue